### ISSUE TITLE
Add support for Gtk+ 3.20 and/or 3.22

### ISSUE THREAD

**Issue by Beep6581** [Beep6581/RawTherapee#3446]

**Comment by Hombre57** [comment]
Small update : I'm still in the process of solving the bugs to make RT/Gtk3 works with Gtk3.22. Wait for a patch during this week.

**Comment by Beep6581** [comment]
> I tried to get RT works with gtk>=3.20, but I'm about to give up. Still on this issue at the moment.

@Hombre57 what is the problem?

**Comment by Hombre57** [comment]
The RT's theme just doesn't load. I have this warning message :
`(rawtherapee.exe:3440): Gtk-←[1;33mWARNING←[0m **: Could not load a pixbuf from /org/gtk/libgtk/theme/Adwaita/assets/check-symbolic.svg.
This may indicate that pixbuf loaders or the mime database could not be found.`

Maybe it is related, or not. I hope so, otherwise I don't know why they doesn't load.

**Comment by Beep6581** [comment]
Is there a branch with your progress?

**Comment by Hombre57** [comment]
I'll put them in gtk2-bugfix.

**Comment by Beep6581** [comment]
@vivo75 we are working on the same thing, please coordinate your work with us here. More to follow...

**Comment by vivo75** [comment]
@Beep6581 I've looked at the thing and I'm not feeling able to go further the css thing, this would be the first gtk project for me, at the moment the following things destroy my hopes:
- when searching the internet for changes 3.18 >> 3.20 only new features are mentioned not incompatibilities (or as I call it regressions) especially not how to fix them.
- the widgets creation spawn hundreds of threads and the stack also reach considerable depth, this make debugging/understanding with gdb rather difficult
- it's a lot of time I don't program c/c++ making me a little bit rusty in the area

I would ask for help but I know you are already rather busy and gtk 3.x is not a priority for the upcoming RT-5 release AFAIK (right decision IMHO)

**Comment by Hombre57** [comment]
Thank you for your work @vivo75, I'll take your PR as a resource in order to complete the revamp/update of the CSS file, and I'll certainly delete it when finished.

I've made some progress today, but nothing to commit yet.

What puzzle me the most is that free space of ~25px around the application itself ! I don't know where to change that, if possible at all !

**Comment by vivo75** [comment]
@Hombre57 , I don't experience that, at least using the with both our changes (the one in the pull request)
![screenshot_20161020_023535](https://cloud.githubusercontent.com/assets/60593/19542404/3257cfbe-966e-11e6-816c-4f20db182784.png)

**Comment by Beep6581** [comment]
@Hombre57 
```
* { font-family: sans-serif; font-size: 8px; } /* Probably not a good idea to use *, better to target specifically */
tab { padding: 0; background-color: #000066; }
tabs { background-color: #006600; font-family: }
header { background: #ff0000; font-family: }

paned box { background-color: #00ffcc; }
paned box { padding: 5px; margin: 5px; }
notebook stack { background-color: #ffcc00; padding: 40px; }
```

**Comment by Hombre57** [comment]
@vivo75 Sorry, I forgot to say it was on Windows.

This is what I get when clicking on the _Maximize_ icon : http://imgur.com/a/SM0hR

**Comment by Beep6581** [comment]
@Hombre57 I'd like to merge master into gtk3, but I don't want to cause merge issues for gtk3-bugfix. What do you suggest?

**Comment by Hombre57** [comment]
You can do it, I'll forward the merge to the gtk3-bugfix branch.

**Comment by Beep6581** [comment]
Monday is the last chance to make an October release. @Hombre57 what are you feelings on merging gtk3-bugfix into gtk3 by then? There must still be time to test that the merge does not break things for Gtk+ 3.16-3.18.

I will merge master into gtk3 tonight.

**Comment by heckflosse** [comment]
@Beep6581 @Hombre57 I'll test using 3.18 on Win7

**Comment by heckflosse** [comment]
@Beep6581 I pushed a [new branch](https://github.com/Beep6581/RawTherapee/tree/integration). It is master merged into a copy of gtk3 branch. I didn't want to push directly to gtk3 branch, but you can use it to solve the merge conflicts when you merge master into gtk3 branch

**Comment by Beep6581** [comment]
Thank you @heckflosse , merged and deleted.

**Comment by Beep6581** [comment]
Wrong button.

**Comment by Hombre57** [comment]
Gtk3 branch merged into gtk3-bugfix done. Working on the remaining flaws, however I don't think I'll be able to solve the gap around the application when in full screen mode on Windows (Win7).

**Comment by heckflosse** [comment]
I found a bug : 
![another_bug](https://cloud.githubusercontent.com/assets/13733487/19824275/568eaf72-9d6e-11e6-9fa3-9e88af25d73a.png)

**Comment by Beep6581** [comment]
@Hombre57 regarding font removal, please see:
```
main.cc:341:                cssForced->load_from_data (Glib::ustring::compose("* { font: %1; }", font));
preferences.cc:2087:            cssForced->load_from_data (Glib::ustring::compose("* { font: %1; }", newFont));
splash.cc:51:    fontd.set_absolute_size (12 * Pango::SCALE);
splash.cc:94:    localCSS->load_from_data ("GtkTextView { font: monospace 8; }");
```

**Comment by Beep6581** [comment]
The ThresholdSelector has a white frame, it does not look good:
![malselm](https://cloud.githubusercontent.com/assets/5844619/19836698/d758c7e2-9ea8-11e6-8ea7-0c0b976c9a82.png)

Fix:
`#ThresholdSelector { border-color: #252525; }`

**Comment by Beep6581** [comment]
Disabled elements are not handled.
[before fix](https://cloud.githubusercontent.com/assets/5844619/19836849/f8921734-9eac-11e6-92b8-9aaa340a0394.png)
[after fix](https://cloud.githubusercontent.com/assets/5844619/19836857/1ba9c05a-9ead-11e6-8c2c-e5f934efef07.png)

Fix:
`*:disabled { color: #666666; }`
or
`label:disabled, check:disabled, radiobutton:disabled, checkbutton:disabled { color: #666666; }`

**Comment by Beep6581** [comment]
Meta tab, I cannot resize the columns:
[problem](https://cloud.githubusercontent.com/assets/5844619/19836871/6e814596-9ead-11e6-8382-5d439134722a.png)

**Comment by heckflosse** [comment]
On my old Sabayon Laptop with 1366x768 screen using gtk3 3.20.1 on gtk3-bugfix branch I can not resize left or right panel. Works fine on master branch using gtk2.
Screenshot follows in a few minutes

**Comment by heckflosse** [comment]
Threshold selector points are not visible in 3.20.1
![malselm](https://cloud.githubusercontent.com/assets/5844619/19836698/d758c7e2-9ea8-11e6-8ea7-0c0b976c9a82.png)

Works fine in 3.18

**Comment by Beep6581** [comment]
Resizing the panels works fine for me using Gtk+ 3.20.9, though the "catchment" area is only 1 pixel wide making it hard to hit it.

**Comment by heckflosse** [comment]
No catch here

**Comment by Beep6581** [comment]
Which one do you prefer?

`#ThresholdSelector { border-color: #252525; }`
![imgur-2016_10_30-22 04 24](https://cloud.githubusercontent.com/assets/5844619/19839991/c469188c-9eec-11e6-8c33-995994a1fdd3.png)

`#ThresholdSelector { border: 1px solid #252525; }`
![imgur-2016_10_30-22 01 24](https://cloud.githubusercontent.com/assets/5844619/19839968/5c69805a-9eec-11e6-8a84-716491a4e52b.png)

`#ThresholdSelector { border: 1px solid #252525; background-color: #161616; }`
![imgur-2016_10_30-22 02 20](https://cloud.githubusercontent.com/assets/5844619/19839972/80e67578-9eec-11e6-8447-fcfb6420656f.png)

`#ThresholdSelector { border: 1px solid #252525; background-color: #AAAAAA; }`
![imgur-2016_10_30-22 03 10](https://cloud.githubusercontent.com/assets/5844619/19839982/9c0bb3fe-9eec-11e6-968d-311809b85786.png)

**Comment by Beep6581** [comment]
My vote: 1, 4, 2, 3.

**Comment by heckflosse** [comment]
4

**Comment by Hombre57** [comment]
@Beep6581 @heckflosse I'm working in this too, I'd like to display the sliders the same way than the standard sliders (been working on this the whole w.e., due to a silly bug).

**Comment by Beep6581** [comment]
@Hombre57 yes, just CSS as shown. Please update me on your status as I intended to release RT5 today and so merge gtk3-bugfix into gtk3.

**Comment by Hombre57** [comment]
Well, Gtk3.20 smells like :shit: with RT, so no, it's not ready yet. I try to solve a nasty bug of the Retinex > Transmission curve, which doesn't resize the Frame they belong to correctly when folding/unfolding the curve.  :rage:

I spent the last few days on Linux to hope for better debug options (and so was away from my emails), but without success.

> How do you mean?

I mean put the Adjuster's slider in the Threshold selector (smaller ones). That would create a more uniform look.

Beside this, the margin around the top window, when maximized and on Windows, is still not solved, unless we're okay to not being able to resize it.

**Comment by Beep6581** [comment]
Looks like this could drag on for some time, and stress won't help anyone. We could release RT5 with "experimental support for Gtk+ 3.20-3.22", saying that 3.16-3.18 are recommended. The patch snippets from this issue should be committed first into gtk3-bugfix, and that merged into gtk3. The remaining problems can be solved for 5.1. Would you prefer this, or would you prefer I wait for a few days?

Further, because the CSS file for 3.16-3.18 is not compatible with 3.20-3.22, we could rename the CSS files to "RawTherapee GTK316-318.css" and "RawTherapee GTK320-322.css" and leave it up to package maintainers to rename the correct file to match the version of Gtk+ they're building against - @sguyader @gaaned92 @mbajor @dhor

What are your thoughts?

**Comment by Hombre57** [comment]
I'll let the package maintainer decide on how to rename the CSS files, but yes, we should provide both for some time.

There is a deadline that is already crossed regarding distributions, sorry for that, but I think the 3.22 version is really near the corner if we accept unresizable windows on Windows and the little graphical glitch in the Retinex tool (all OS).

**Comment by vivo75** [comment]
it's possible to load a different css based on the gtk version actually installed on the system?
timing example, user install the packages in this order:
gtk-3.18
Rawtherapee
gtk-3.20
At the moment the css for 3.18 will be loaded by RT breaking things.
This is also a very plausible scenario, it's difficult to think that packagers will update at the same time RT _and_ gtk,  maybe on newly installed machines.

Also the names of the css should be "RawTherapee GTK316-318.css" and "RawTherapee.css", just in case it will continue to work after 3.22

BTW I've tried to find something useful into the [Breeze theme](https://quickgit.kde.org/?p=breeze-gtk.git&a=blob&h=3881e5f9214c9c4a665d1778058d79535249d630&hb=2fb123e35cf9c7710fb0e34f34bb3b33163e9aee&f=kconf_update%2Fmain.cpp) which has different css for gtk-3.20 but it seem it's only looking at the on disk configuration.

**Comment by Hombre57** [comment]
First, I'd be for removing the "Use system theme" option, because it'll be very ugly, as RT make use of CSS selectors to have a correct look. Do anyone agree on this ?

Then **is there a way are run time to check the Gtk version** ? In this case, we could load the correct CSS theme automatically. The Toowablue has not been updated for 3.22 (or even 3.18 ?), so we shouldn't bundle it at first if that's the case. Adding a suffix to the CSS theme, following this template (xxx-gtkMajor.gtkMinor.css, eg: Rawtherapee-3.18.css) will let RT use the correct CSS file.

**Comment by Hombre57** [comment]
Found it : `gtk_get_major_version ()` and `gtk_get_minor_version` will do the job :)

**Comment by vivo75** [comment]
https://developer.gnome.org/gtk3/stable/gtk3-Feature-Test-Macros.html

**Comment by gaaned92** [comment]
@Beep6581 : not being able to build on Windows10 with GTK+ 3.18 (32 and 64 bit), and stuck on 3.22, I   vote for the second proposal. But no problem if the first proposal is adopted. (should not be difficult to rename during the build process)
Windows 32bits builds users will not be surprised if Rawtherapee 5.0 is delivered only with GTK2 flavor.
For 64 bit users,   @sguyader  is able to provide GTK3.18 builds.(hoing he will not update MSYS2!)
So I think that for windows users it is possible to wait 5.1 to switch to 3.22

@Hombre57 I never used the system theme so OK to suppress it

**Comment by heckflosse** [comment]
@Hombre57 

> The Toowablue has not been updated for 3.22 (or even 3.18 ?)

Toowablue works fine on Win7 using gtk 3.18

**Comment by heckflosse** [comment]
@Hombre57 :+1: to remove `Use system theme`

**Comment by heckflosse** [comment]
@Hombre57 @gaaned92 @sguyader 
Imho for Windows builds we should use what's best for rt windows builders and also works fine. On Windows we have the quite comfortable situation (compared to Linux) that we have to maintain only 2 release builds (one for win32 and one for win64). For that reason I would rely on André and Sébastian to decide and I won't care about Linux gtk3 deadlines in that case (at least not for the rt5 release)

**Comment by Hombre57** [comment]
@heckflosse 

> On my old Sabayon Laptop with 1366x768 screen using gtk3 3.20.1 on gtk3-bugfix branch I can not resize left or right panel.

Is it still the case ?

**Comment by heckflosse** [comment]
@Hombre57 

> Is it still the case ?

I did not try today, but yesterday it was the case with latest gtk3-bugfix version. Shall I try again?

Ingo

**Comment by Beep6581** [comment]
@vivo95

> user install the packages in this order:
gtk-3.18
Rawtherapee
gtk-3.20
At the moment the css for 3.18 will be loaded by RT breaking things.

RawTherapee built against 3.18 will require and use 3.18 even if you additionally install 3.20 in a slot (which in Gentoo is not possible - you can have Gtk+ 1.* alongside 2.* alongside 3.*, but only one of each). The loaded CSS has to match the Gtk+ version RawTherapee was _built against_.

> just in case it will continue to work after 3.22

Gtk+ devs wrote that the 3.22 API will be rock solid for years, that's why it makes sense to support that. There will not be a newer 3.* version than 3.22.*.

@Hombre57 

> First, I'd be for removing the "Use system theme" option, because it'll be very ugly, as RT make use of CSS selectors to have a correct look. Do anyone agree on this ?

I don't mind removing "Use system theme". It worked well with Gtk2, but looks bad with Gtk3.

> The Toowablue has not been updated for 3.22 (or even 3.18 ?)

`TooWaBlue.css` works fine with 3.16.* and 3.18.*.
`RawTherapee.css` from `gtk3` works fine with 3.16.* and 3.18.*.
`RawTherapee.css` from `gtk3-bugfix` works fine with 3.20.* and 3.22.*.

**Comment by Hombre57** [comment]
> I did not try today, but yesterday it was the case with latest gtk3-bugfix version. Shall I try again?

Yes please, I don't have problem to resize the L/R panels with Gtk3.22.

**Comment by Beep6581** [comment]
I can resize the panels using 3.20.9, though the "catchment area" is only 1 pixel wide making grabbing the resize handle quite difficult.

**Comment by gaaned92** [comment]
Branch: gtk3-bugfix
    Version: [4.2.1358]
    Changeset: 3ae608d5caea5de3106a705dca495bbfe9b7db25
    Compiler: gcc 6.2.0
    Processor: generic x86
    System: Windows
    Bit depth: 64 bits
    Gtkmm: V3.22.0
My first observations:
1- in full screen, the margin around the top window is deactivated (you cannot click on another window)
2-the middle button on the title bar should show two squares superposed to indicate that we can reduce the window.
3- if you displace the window, then you can change the size of it . but the behaviour becomes erratic
4-  in file browser, not possible to reduce right panel under 10 cm and left under 7 cm
5- about the same for editor, Right panel must be 10 cm to display whole widgets
6 all widgets seem to me much too large compared to GTK2 in the exposure panel. It seem much better in other panels
7 zoom: when you zoom in, the red rectangle in the navigator (left panel) is not coherent with what is shown

[see screenshot](http://imgur.com/96pO7PJ)

@Hombre57 : can you upload a zip of your build so that I can verify nothing is missing in my build.  hope that can help you in this hard work
André

**Comment by heckflosse** [comment]
@gaaned92 

> 7 zoom: when you zoom in, the red rectangle in the navigator (left panel) is not coherent with what is shown

I think the navigator is correct here but the preview in editor is partly hidden by the right panel

**Comment by Hombre57** [comment]
I did mentioned issue 3446 in the commit, without hash tag tough, but there's no entry in there. So you can see commit 965cadb for the last improvement.

Last thing to do and we'll be ready :
   - [ ] creating the mechanism to load the correct css file
   - [ ] address the Profile Selector button size (if possible)

**Comment by gaaned92** [comment]
@Hombre57 
version 4.2.1359
1- suppresion of gap ok. 
2- ok No problem for me with limited possibility to reduce size of window
3- ok possible to drag the reduced window without problem
4- not possible to reduce size of right and left panel. Not enough space left for displaying image
5- and 6- with the right panel minimum size, some informations of the exposure panel are hidden (right part of widgets, processing profile selector name). if I enlarge the right panel to see all information, there is not enough place to display image correctly.
 in exposure panel, widgets width  should be adapted to right panel dimensions
7- zoom ok

On GTK2, I use smaller font. It is not possible here to choose the font size in the preferences.
I have some difficulty to choose the processing profile.I cannot always select the bundled or custom profiles. the  behavior is not deterministic.

**Comment by Hombre57** [comment]
@gaaned92 Just for my information, what is you screen size in pixels ? Seems to be small.

> On GTK2, I use smaller font. It is not possible here to choose the font size in the preferences.

There's a way to configure the Gtk apps globally, but I have to find that again.

**Comment by Beep6581** [comment]
@Floessie wrote in https://github.com/Beep6581/RawTherapee/issues/3482#issuecomment-257804324

> there is only minimal space between the tab titles.

To increase tab separation (tested only in SETM - Vertical)
`#MainNotebook > header tabs tab { margin: 1em 0 1em 0; }`

**Comment by Hombre57** [comment]
That's how things are done in Gtk3, see the Notebook class to see the sub elements. We can't name them ourselves in this case. Search for "*CSS Nodes*" here : https://developer.gnome.org/gtk3/stable/GtkNotebook.htm

**Comment by dhor** [comment]
Hi can you please remove me (dhor@mac.com) from your mailing list, this is
nothing to do with me.
Thanks

2016-11-02 11:35 GMT+01:00 Jean-Christophe <notifications@github.com>:

**Comment by Beep6581** [comment]
Hi @dhor I removed your nick mention from a previous comment, though I doubt that will change anything. I don't think it's possible for anyone other than yourself to remove yourself from getting notifications about updates to this issue. Try "unwatch" or "unstar" in the top-right corner, or "unsubscribe" in the menu on the right. Are you not Dariusz Duma?

**Comment by dhor** [comment]
Hi, I dont know where I was subscribed to this... and I am not Dariusz

2016-11-02 12:55 GMT+01:00 Beep6581 <notifications@github.com>:

**Comment by Beep6581** [comment]
I see you're using email. The buttons I referred to should be visible if you go to https://github.com/Beep6581/RawTherapee/issues/3446
I would assume if you're getting emailed then the email should also describe how to unsubscribe, usually in the footer.

**Comment by Beep6581** [comment]
Things are looking good.

A few final tweaks in https://github.com/Beep6581/RawTherapee/pull/3508

1. The background color of the About and Preferences window was not set.
2. Some of the About window's tabs have a font size hard-coded in pt, resulting in text in some tabs looking significantly bigger than in other tabs and in other programs. Removing the font selector means that font sizes should be inherited from the user's system.
https://i.imgur.com/0VM0M2T.png
https://i.imgur.com/v7Xs3HO.png
3. Preferences/About window About/OK/Cancel buttons are too small, they lack padding and margins. But the correct fix is not with CSS, see points 4 and 5.
4. There is an empty action_box below the "Close" button:
https://i.imgur.com/iI4Jk8Q.png
The GTK Dialog has a bottom area called the action area where "Close", "OK", "Apply" and "Cancel" buttons should go. Doing so also fixes the button sizes.
https://i.imgur.com/TJstAhq.png
5. Same as 3, but for Preferences:
https://i.imgur.com/CI3w38D.png
https://i.imgur.com/1J7cqz6.png
6. Notebook buttons are inconsistent, those in Preferences and About are too small and too close together:
Too small and too close together https://i.imgur.com/mGXyzr9.png
7. #3495 "Gtk+ >=3.20 Separators in Filmstrip toolbar".
8. One important (to me) fix is that the paned separator ("catchment area" where I can grab and resize panels) was only 1 pixel wide, making resizing those panels difficult or impossible. These: https://i.imgur.com/Q2Uygc3.jpg

**Comment by Floessie** [comment]
> Things are looking good.

Indeed, things almost look familiar. One thing left is the vertical overall progress bar, which is relatively thin and fills top-down instead of bottom-up. That's one strange general experience with GTK3: The (scroll/progress) bars became thin to save space, while almost every other element got more padding, so effectively reducing the number of items on screen. I still have to get used to that...

Very good job, @Hombre57 and @Beep6581!

**Comment by Floessie** [comment]
The preferences and fullscreen buttons (bottom left) changed places. Is that intentional?

When directly comparing `master` against `gtk3-bugfix`, some things stand out:

1. Fonts are bigger in GTK3.
2. There is a broader horizontal padding around the vertical tab texts in GTK3.
3. The "double padding" on the sides of the image area looks a bit asymmetric in both GTK2 and GTK3. E.g. the history panel looks like it has more padding to the right (thumb area, image area) than to the left (vertical tabs, overall progress, preferences button, fullscreen button).

Best
Flössie

**Comment by Beep6581** [comment]
@Floessie Font size in `gtk3-bugfix` is inherited from the system theme so that all programs look consistent. We will need to find how Windows users should change that and include this info in the release notes.
https://i.imgur.com/sL5bJoQ.jpg

**Comment by innir** [comment]
Wow, I did a build of 0ab931c287e9a725fe989b3f687f8cdca18655f4 last night and I must say it looks really good now. Great work, thanks!

**Comment by Hombre57** [comment]
I converted the filenames for the new version handling mechanism, but don't know what to do of TooWaBlue.css. Actually, this theme is not supported, so it's not listed in the Preferences' combobox.

@Beep6581 As said by @heckflosse on IRC, does any distribution still use Gtk3.18 version ? i.e. should we support it ? Since you're in the air right now and can't answer, I'll still update the 3.18 version.

**Comment by noetzelj** [comment]
This message was created automatically by mail delivery software.

A message that you sent could not be delivered to one or more of its
recipients. This is a temporary error. The following address(es) deferred:

  jnuts.pilot@gmail.com
    Domain noetzel.net has exceeded the max emails per hour (265/250 (106%)) allowed.  Message will be reattempted later

------- This is a copy of the message, including all the headers. ------
Received: from o1.sgmail.github.com ([192.254.114.176]:16470)
	by whub40.webhostinghub.com with esmtps (TLSv1.2:ECDHE-RSA-AES128-GCM-SHA256:128)
	(Exim 4.87)
	(envelope-from <bounces+848413-25bb-jim=noetzel.net@sgmail.github.com>)
	id 1c8QZ6-00052T-9i
	for jim@noetzel.net; Sun, 20 Nov 2016 03:45:44 -0800
DKIM-Signature: v=1; a=rsa-sha1; c=relaxed; d=github.com; 
	h=from:reply-to:to:in-reply-to:references:subject:mime-version:content-type:content-transfer-encoding:list-id:list-archive:list-post:list-unsubscribe; 
	s=s20150108; bh=tOPftGtIl3nsERdsSPmZFdtjzlk=; b=KtjxdoZXNiBcTPDp
	ukIbH488PKgidOGLs1bMXlOVjod5gmHe7Nrj9Pq75otWlGCK5MgoVbtAQwbpyhCo
	rljRcfn1PxQsvbvJ4O3oy/wZnYStW863yr/c/sa8R/ZUPY29MYDid9oTpMWnp/FB
	YdxwQCr7t95dfq+tVHVwOCOUILg=
Received: by filter0563p1mdw1.sendgrid.net with SMTP id filter0563p1mdw1-16577-58318CE5-60
        2016-11-20 11:45:41.966600283 +0000 UTC
Received: from github-smtp2a-ext-cp1-prd.iad.github.net (github-smtp2a-ext-cp1-prd.iad.github.net [192.30.253.16])
	by ismtpd0003p1iad1.sendgrid.net (SG) with ESMTP id lR8Oy2XRQmyEPHrl8GwraQ
	for <jim@noetzel.net>; Sun, 20 Nov 2016 11:45:42.017 +0000 (UTC)
Date: Sun, 20 Nov 2016 03:45:41 -0800

**Comment by Floessie** [comment]
@Hombre57 The vertical overall progress bar still fills top to bottom, which looks a bit odd. Is that intended?

**Comment by Hombre57** [comment]
@Floessie
No, it wasn't intended. `set_orientation` has new `enums` in Gtk3. I just found the `set_inverted` method to invert the default fill direction. Committing a patch now.

**Comment by Beep6581** [comment]
@Hombre57 most distributions still only provide GTK+ 3.18 or older, and only a few have the option of providing >=3.20. Those that do offer >=3.20 usually are of the rolling-release or snapshot style (e.g. OpenSUSE "Tumbleweed" or Debian unstable "Sid"), not the stable versioned style. RawTherapee supported <=3.18 well before we started with `gtk3-bugfix`, and after merging `gtk3-bugfix` into `gtk3` it should support both GTK3 ranges using two CSS files.

Could you update me on the status?

**Comment by Hombre57** [comment]
Hi @Beep6581 , nice to see you back !

The status is that I've made good progress in Gtk3.18 support, but it's not finished, and TBH more difficult than expected. I thanks @heckflosse for giving me hand over his PC through RPC, and I'll certainly have to accept his offer of getting his laptop that is installed with 3.18. I've tried to revert to Gtk3.18 but MSYS and Manjaro are rolling release and don't provide that package anymore, unless you have a trick to do so ?

**Comment by Beep6581** [comment]
If they don't provide that package then reverting manually will probably break your system, because the other binary packages will have been built against the newer version. Easiest option is to install VirtualBox and any Linux distro which uses 3.18:
https://www.virtualbox.org/
http://releases.ubuntu.com/16.04/ubuntu-16.04.1-desktop-amd64.iso

Do I understand correctly that you aim to get a single CSS file to support both GTK+ <=3.18 and >=3.20?
Are you doing this because once `gtk3-bugfix` is merged into `gtk3` then the old 3.18-friendly CSS file will no longer work?

**Comment by Hombre57** [comment]
Yes, aftet merge, the actual css theme will not work anymore as @heckflosse has experienced. However @heckflosse, could you confirm that you're using 3.18 and not a previous version (I don't remember what you told me on IRC, sorry).

The automatic theme selection mechanism depending on Gtk version is already done in bugfix branch.

**Comment by heckflosse** [comment]
@Hombre57  V3.18.0

**Comment by Hombre57** [comment]
It appears that we can't make anything good out for the CSS file alone for Gtk3.18, which was unexpected for me. I'll revert my change done in the code but enclosed in `#ifdef` statements, so that they are restored for Gtk3.18 only (does Gtk3.19 exist btw?). That way, when we'll be ready to drop support of Gtk3.18, we'll only have to make some code cleanup. I'll do that tonight and hope to commit tomorrow in the evening.

For issue #3494 and #3261, which might be the same bug, I have still no idea of what's going and won't be solved for RT5. Users will still be able to make it working by moving the pane's separator, but that's not what I'd call a satisfying user experience.

**Comment by Hombre57** [comment]
@TooWaBoo I see that you're very experienced with Gtk's CSS files (way more than me). Would you mind helping me with the Gtk3.18 theme ? e.g. this rule doesn't work :
```
#ExpanderBox2 GtkFrame { /* > border */
    background-color: #414141;
    border: 1px solid #373737;
    border-radius: 4px;
    margin: 3px;
    padding: 3px;
}
```

And I don't manage to set a `background-image` to the `GtkCheckButton` :
```
GtkCheckButton.check {
    background-image: linear-gradient(#343434, #2E2E2E, #292929);
}
```

Ideally, you should try with the CSS them from the gtk3 branch copied - and renamed - into the gtk3-bugfix branch.

**Comment by TooWaBoo** [comment]
The background is covered from the checkbox **icon**. You have to remove or replace the icon with an transparent one.
Try this:
```
.check {
    -gtk-icon-source: none;
    background-image: linear-gradient(#343434, #2E2E2E, #292929);
 }
```

**Comment by Hombre57** [comment]
I think I have found the problem of the wrong display of some GUI elements. With a bit of luck, it should solve issue #3261 and #3494 as well. Lesson 2 : RTFM ; Lesson 1 : FTFM (Find That Fu**ing Manual)

**Comment by Beep6581** [comment]
@Hombre57 could you PUTTFM (Point Us To That Friendly Manual)? ;)

**Comment by Hombre57** [comment]
https://developer.gnome.org/gtkmm-tutorial/stable/sec-custom-widgets.html.en

Google brought me to that page.

**Comment by Beep6581** [comment]
@Hombre57 I looked again into the catchment area of the separators which you grab with the mouse to resize the panels, emphasized here https://i.imgur.com/Q2Uygc3.jpg
The problem currently is that there is a lot of dead space around this catchment area, you can clearly see it here https://i.imgur.com/0jjQjAN.png Look how much non-green space there is around the green lines, especially on the right side of the line in the toolbox. We could reduce the amount of mouse hunting required by removing that dead space. Also in the File Browser https://i.imgur.com/FhlRhaK.png on the left, there is even a small gap in the green line in the bottom-right corner of the big minus button in the Places panel.

First we can recover a few pixels from the main menu bar on the left in Vertical Tabs mode https://i.imgur.com/Yh6d4f0.png
https://i.imgur.com/LuQ26cW.png

Next we can recover some space from around the #PlacesPaned and make the green lines touch without gaps
https://i.imgur.com/CLRkPuY.png
(this only has an effect in the File Browser)

I ran out of time to dig further, so the quick rough version for the rest:
Editor tab https://i.imgur.com/qdYPBcC.png
Remove padding from frames https://i.imgur.com/supHCVJ.png

Remove the left margin from #MyExpander to get rid of that dead space to the right of the green line in the toolbox https://i.imgur.com/s92LOUJ.png

That would be generally enough. With all the saved space, we could even increase `paned > separator` from 1px to 2px to make resizing those panels easier if someone needs it.

All CSS here:
```
paned > separator {
	border: 1px solid #004400;
}

/* Preferences / Editor / Queue / File Browser bar */
#MainNotebook header {
	border: 0;
	padding: 0;
}

#PlacesPaned {
	padding: 0px 0px 5px 5px;
}

frame {
	padding: 0;
}

#MyExpander {
	margin-left: 0;
}
```

**Comment by Hombre57** [comment]
@heckflosse Could you tell us if the Gtk3.18 theme is _acceptable_ ? I don't think we should spend too much time to fine tune it (we're not able to do it because 3.18 doesn't goes as far as 3.22 regarding CSS).

@Beep6581 2px on the left of the History, Navigator and Snapshots frame is really not enough, we can't clearly distinguish the elements IMHO. And if you remove the left margin from `#MyExpander`, the tools has no space at all from the Batch Tool Panel.

I'm trying to find a solution on my side too.

**Comment by heckflosse** [comment]
> @heckflosse Could you tell us if the Gtk3.18 theme is acceptable ? I don't think we should spend too much time to fine tune it (we're not able to do it because 3.18 doesn't goes as far as 3.22 regarding CSS).

I will test now...

**Comment by TooWaBoo** [comment]
@Beep6581, @Hombre57 
This is what I'm doing to get a 6px resize area arround the main window:
```
/*Main window*/
window:not(.popup):only-child > decoration {
--> border: 1px solid @winHeaderbar;
    border-radius: 0;
    margin: 0;
    box-shadow: none;
--> padding: 5px;
    background-color: @bg-light-grey;
}
/*expand titlebar to the left and right and move to the top to cover the padding part*/
window:not(.popup):only-child > headerbar {
--> margin: -5px -5px 0;
}
```

To get a **5px** separator, with a **9px** resize area, you can do this:
```
paned.horizontal > separator {
    background-color: transparent;
--> border-left: 4px solid @bg-light-grey;
--> border-right: 4px solid @bg-light-grey;
--> margin: 0 -2px;
    padding: 0;
}
```
I hope this helps

**Comment by Hombre57** [comment]
@TooWaBoo I've seen your comment too late, I'll try that tomorrow.

@Beep6581 I've pushed some changes that goes toward what you want I hope. I hope you don't mind that I pushed directly.

**Comment by TooWaBoo** [comment]
@Hombre57 
Another trick

A separator with a thin double line
```
paned.vertical > separator {
    background-color: @bg-dark-grey;
    border-bottom: 4px solid @bg-light-grey;
    border-top: 4px solid @bg-light-grey;
    margin: -1px 0;
    padding: 1px 0;
}
```
The border left is the windows padding. The border right is the horizontal separator. And the double line separator is like described above
![unbenannt](https://cloud.githubusercontent.com/assets/13800920/21084951/32a57b3a-c00b-11e6-896a-94b98aeebd65.png)

I've attached the latest beta theme, so can examine the whole code.

[TooWaBlue-2.08-GTK3-20_.txt](https://github.com/Beep6581/RawTherapee/files/644923/TooWaBlue-2.08-GTK3-20_.txt)

**Comment by TooWaBoo** [comment]
btw.
Because we are not using a native window, I think we have to do the whole window control.

These are the steps for the gui file
1. Replace the Maximize button with a Restore button when maximezed
2. Assign a class .maximized to the window when the window is maximized
3. Remove the class .maximized when the window is not maximized
4. Assign a class .fullcreen to the window when we are in fullscreen
5. Remove the class .fullcreen when the window is not in fullscreen

With these modifications we can add or remove the decoration of the window in the theme file and the window resize problen is gone.

**Comment by Beep6581** [comment]
@TooWaBoo unfortunately even after removing the comments and replacing the color definitions the code you pasted does not work here in GTK+ 3.22.3 in KDE. The double line separator looks nice. Ideally I'd like a single, thick dashed line to show the separators, but I didn't manage to make one.

@Hombre57 your latest commit looks and behaves well! I consider the separator problem solved.

**Comment by Beep6581** [comment]
Ha, I got dashed lines:
https://i.imgur.com/S6psXtc.png
```
paned > separator {
	border:none;
	border-top:3px dashed #888888;
	border-left:3px dashed #888888;
	color:#484848;
}
```
But we don't need them - the way it is in commit e6aff9c looks nice.

**Comment by TooWaBoo** [comment]
> @TooWaBoo unfortunately even after removing the comments and replacing the color definitions the code you pasted does not work here in GTK+ 3.22.3 in KDE.

What does this mean? When you try the TooWaBlue theme in RT, doesn't it look like the screenshot?

> The double line separator ... but I didn't manage to make one.

I guess, another style is overwriting it. GTK won't have the same rules like in regular css.

**Comment by Beep6581** [comment]
> What does this mean? When you try the TooWaBlue theme in RT, doesn't it look like the screenshot?

I tried using your code snippets on top of `RawTherapee-GTK3-20_.css` via GTK Inspector, replacing the definitions as required. I didn't try TooWaBlue.

> I guess, another style is overwriting it. GTK won't have the same rules like in regular css. Not only the order counts, it depends how complex the selector is defined. It makes a different if you use only e.g. "paned" or "panet.vertical"

I know, I read the docs. The rest is no longer relevant as the latest commit looks and works well.

**Comment by heckflosse** [comment]
I just tested gtk3-bugfix on Win7/64 using gtk 3.18.6. Works and looks good

**Comment by TooWaBoo** [comment]
@Hombre57 
Is it possible to get an ID for Snapshots and Processing profiles?

![1](https://cloud.githubusercontent.com/assets/13800920/21109402/3ab2972a-c099-11e6-8765-2b98c8969b5b.png)

**Comment by Beep6581** [comment]
@TooWaBoo Processing Profiles already have an ID, #ProfilePanel

To add a #Snapshots ID:
```diff
diff --git a/rtgui/history.cc b/rtgui/history.cc
index a8942fb..a1ad451 100644
--- a/rtgui/history.cc
+++ b/rtgui/history.cc
@@ -112,6 +112,7 @@ History::History (bool bookmarkSupport) : blistener(nullptr), tpc (nullptr), bmn
     bscrollw->set_size_request (-1, 45);
 
     Gtk::Frame* bmFrame = Gtk::manage (new Gtk::Frame (M("HISTORY_SNAPSHOTS")));
+    bmFrame->set_name("Snapshots");
     Gtk::VBox* bmBox = Gtk::manage (new Gtk::VBox ());
     bmFrame->add (*bmBox);
     bmBox->pack_start (*bscrollw, Gtk::PACK_EXPAND_WIDGET, 4);
```

**Comment by TooWaBoo** [comment]
@Beep6581 
Thanks 👍

**Comment by Hombre57** [comment]
Patch committed, name added to the History panel as well.

**Comment by Beep6581** [comment]
Great! I'll polish up the release notes then.

**Comment by heckflosse** [comment]
I just detected that the size and position of rt window is not remembered when closing rt.
It reads the values from options file on start, but it seems it doesn't save them on close.
Win7/64 gtk3.18.6

**Comment by Floessie** [comment]
@heckflosse There are some new `GdkWindowState`s which mess things up, especially `GDK_WINDOW_STATE_FOCUSED`: [GTK 2](https://developer.gnome.org/gdk2/stable/gdk2-Event-Structures.html#GdkWindowState), [GTK 3](https://developer.gnome.org/gdk3/stable/gdk3-Event-Structures.html#GdkWindowState).

Here is a patch serving as a base for @Hombre57:

```patch
diff --git a/rtgui/rtwindow.cc b/rtgui/rtwindow.cc
index 3b9995b..f00eb63 100644
--- a/rtgui/rtwindow.cc
+++ b/rtgui/rtwindow.cc
@@ -385,11 +385,12 @@ void RTWindow::on_realize ()
 
 bool RTWindow::on_window_state_event(GdkEventWindowState* event)
 {
+    const int new_window_state = event->new_window_state & ~GDK_WINDOW_STATE_FOCUSED;
 
-    if (!event->new_window_state) {
+    if (!new_window_state) {
         // Window mode
         options.windowMaximized = false;
-    } else if (event->new_window_state & (GDK_WINDOW_STATE_MAXIMIZED | GDK_WINDOW_STATE_FULLSCREEN | GDK_WINDOW_STATE_ICONIFIED)) {
+    } else if (new_window_state & (GDK_WINDOW_STATE_MAXIMIZED | GDK_WINDOW_STATE_FULLSCREEN | GDK_WINDOW_STATE_ICONIFIED)) {
         // Fullscreen mode, save this mode even when window is iconified (minimized) to allow easier restore, not the best solution though...
         options.windowMaximized = true;
     }
```

HTH
Flössie

**Comment by heckflosse** [comment]
@Floessie Thank you. Your patch fixes it. :+1:

**Comment by Beep6581** [comment]
> Because we are not using a native window,

I am curious what the reason for that is.

**Comment by TooWaBoo** [comment]
> I am curious what the reason for that is.

The window decoration is outside of the window. If you don't switch it off it looks like this in maximize mode.
![unbenannt](https://cloud.githubusercontent.com/assets/13800920/21236813/c58c0c86-c2fc-11e6-89e2-80849457b416.PNG)

The border around the window is the window decoration. If you remove it, than you don't have a resize area.
The maximize button works, but there is no restore button. Without using a native window the window control has to be done bei RT.

**Comment by Beep6581** [comment]
By "resize area" do you mean a window border handle which you can grab to resize a not-maximized window?

**Comment by TooWaBoo** [comment]
Yes, but this doesn't mean the border is visible, that's why I'm calling it area. :-)

**Comment by Hombre57** [comment]
>> Because we are not using a native window,

> I am curious what the reason for that is.

Gtk uses Client Side Decoration (CSD) i.e. rendering entirely done by Gtk, while Gtk2 was using Server Side Decoration (SSD), i.e. rendering done by the OS' window manager.

At first I thought that it was a regression, but after reading what it enables, it's a nice move IMHO. I still have to get the final point on this, but at the moment it seem that Gtk can now handle both possibilities. Whereas the GNOME applications or a particular one does use CSD or SSD by default is up to the OS, if I've correctly understood. I still have to find how this is defined. It looks like it is set in the settings file ( `etc/gtk-3.0/settings.ini`, or perhaps `etc/gtk-3.0/.settings` on Linux.

https://blogs.gnome.org/mclasen/2014/01/13/client-side-decorations-continued/
https://blogs.gnome.org/mclasen/2014/01/20/even-more-client-side-decorations/ (not read this one yet)

By looking at the _CSS nodes section_ here : https://developer.gnome.org/gtk3/stable/GtkWindow.html , the `csd` and `ssd` can be tuned separately. I tried to add the `ssd` class and remove the `csd` class for the main window : the `ssd` has correctly been added but the `csd` class is still used, and the look of the window is still the same, so this doesn't work.

Altering the `window.maximized > decoration` as suggested by @TooWaBoo doesn't work here (Win7).

I've committed a patch that adds a 4px border around windows and dialogs as shown by @TooWaBoo, but for both windowed and maximized mode (commit 11dfc71). This is an acceptable trade to me until it we find a better solution. I'll test this theme on Linux tomorrow. At least we can resize the windows on Windows now.

**Comment by Beep6581** [comment]
My `/etc/gtk-3.0/settings.ini`:
```
[Settings]
gtk-theme-name = Adwaita
gtk-icon-theme-name = gnome
gtk-cursor-theme-name = Adwaita
```

> This is an acceptable trade to me until it we find a better solution.

Agreed. The window could be resized on Linux (KDE Plasma) before and after the patch - I could see no difference.
https://i.imgur.com/AaxGM5f.png
The window above is 1901f3a, underneath 11dfc71.

**Comment by TooWaBoo** [comment]
> I've committed a patch that adds a 4px border around windows and dialogs

@Hombre57 
 For a dialog , there is no need to switch the decoration off. It never becomes maximized.
Add a 1px border with the color of the headerbar looks better and adds another 1pixel.
```
window > decoration {
    box-shadow: none;
    margin: 0px;
->  border: 1px solid <headerbar color>;
    border-radius: 0;
    padding: 3px;
    background-color: #262626;
}
```

**Comment by Hombre57** [comment]
@TooWaBoo The headerbar bg color is `#262626` so we'll see no difference here, excepted that there will be one more pixel around the window.

**Comment by TooWaBoo** [comment]
@Hombre57 
I've found this on the i-net. Maybe it helps.

http://stackoverflow.com/questions/37035936/how-to-get-native-windows-decorations-on-gtk3-on-windows-7-and-msys2
> Actually, Gtk+3 been compatible with Windows 10 and you can use the native windows theme by making this your default theme.
> gtk-theme-name = MS-Windows

**Comment by Hombre57** [comment]
Thanks @TooWaBoo , I came across this too, but it look like this is a theme that mimics the native theme but doesn't use the host window manager theme. They even say the it create Win7 windows on Win10 OS. Anyway, I'm fine with actual RT's theme now that you gave me the solution on how to make the window resizable. Our theme is based on Adwaita so it's better to keep `gtk-theme-name = Adwaita`

**Comment by TooWaBoo** [comment]
@Hombre57 
I've found a better and nicer way for the window issue. I've only add a solid 5px border around the window. This should be cut off in Linux and would be a nice **workaround** (see screenshots below). Can you provide some screenshots in Linux if everything looks like I'm expecting, please?
[TooWaBlue-no-dd-win-GTK3-20_.txt](https://github.com/Beep6581/RawTherapee/files/658966/TooWaBlue-no-dd-win-GTK3-20_.txt)
In rtwindow.cc, I saw that you are dealing with the window states maximize and fullscreen. Is it a big job to assign these states to the window as a class? If it can done quickly, I can do some test with the window decoration to switch it on and off on the fly. This would solve the everything.

![1](https://cloud.githubusercontent.com/assets/13800920/21288663/7715547a-c489-11e6-9792-6ad5e9aff62d.PNG)
![2](https://cloud.githubusercontent.com/assets/13800920/21288665/7b24ee2c-c489-11e6-9f40-d57438d26720.PNG)
![3](https://cloud.githubusercontent.com/assets/13800920/21288666/81b34978-c489-11e6-9383-ce990d2a73a0.PNG)
![4](https://cloud.githubusercontent.com/assets/13800920/21288667/84078b9e-c489-11e6-8219-b725ba6b19e3.PNG)
![5](https://cloud.githubusercontent.com/assets/13800920/21288670/8c3a3e42-c489-11e6-96aa-152aca600721.PNG)

**Comment by Hombre57** [comment]
@TooWaBoo Sorry, I don't understand what you mean by _assign these states to the window as a class_ If you mean assigning and removing a class when switching state, then yes it's easy.

**Comment by TooWaBoo** [comment]
> ... assigning and removing a class when switching state, then yes it's easy.

Yes, that's what I mean. Sorry about my bad English.
To get a perfect window control, these states would be necessary:
maximized, fullscreen, active and maybe minimized (but what can you see, when the window is minimized?, hmmm)

**Comment by Hombre57** [comment]
What I don't understand from the documentation is that e.g. `window.maximized` already exist but doesn't seem to be attached automatically, they don't say that we have to assign it ourselves neither... Anyway I'm adding that feature now.

**Comment by TooWaBoo** [comment]
> What I don't understand from the documentation is that e.g. window.maximized already exist but doesn't seem to be attached automatically

That's what I thought too and it didn't work.

**Comment by TooWaBoo** [comment]
@Hombre57 
I think I've found the cause of the window problem. In Windows, when you move a window to a screen border it becomes maximized. I did that with a dialog window. The dialog became maximized, the decoration was automatically removed without doing anything special and the class .maximized was assigned to the dialog.

There must be a misconfiguration for the main window.

**Comment by heckflosse** [comment]
> In Windows, when you move a window to a screen border it becomes maximized.

I can not confirm that on my Win7/64

**Comment by Hombre57** [comment]
I confirm that with my Win7/64 ; try with the Preferences window.

**Comment by TooWaBoo** [comment]
@heckflosse 

You have to move the window, until the mouse pointer touches the screen border.

**Comment by Hombre57** [comment]
Thanks for the hint, it then took me 20mn to find the problem. The great thing is that now we can _unmaximize_ the window ! This should solve the issue on all OS that has Client Side Decoration.

**Comment by heckflosse** [comment]
> You have to move the window, until the mouse pointer touches the screen border.

I confirm that it works with most windows. I tried with current master branch main and preferences window. There it does not work. I didn't try with rt gtk3-bugfix branch because the statement was:
```
In Windows, when you move a window to a screen border it becomes maximized.
```
which implied that it does so for all windows...

**Comment by Hombre57** [comment]
@heckflosse 
> which implied that it does so for all windows...

...which is the case, that's a Windows7 mechanism.

**Comment by TooWaBoo** [comment]
@Hombre57 
In the RT theme, you set this style:
```
window > decoration, dialog > decoration {
    box-shadow: none;
    margin: 0px;
    border-style: none;
    border-radius: 0;
    padding: 3px;
    background-color: #262626;
}
```
This can be deleted. RT handels the decoration automatic.

**Comment by dnw3039** [comment]
I am using the latest gtk3-bugfix Version: 4.2.1443. on Ubuntu 16:10. Rawtherapee theme and font size 10.

When I try to use the tool "reset to default" arrow the right side bar for dragging down the tool panel expands to cover about half the arrow button and makes it hard to accurately hit.  (see the screen pic). I understand the side bar expands to make it more visible. I think it needs to expand a bit less or there needs to be a bit more margin between it and the arrow button.

![screen](https://cloud.githubusercontent.com/assets/7539943/21487242/b20fae0e-cc18-11e6-98a2-6f22c2b5edaf.png)

**Comment by Hombre57** [comment]
@dnw3039 This has been solved in a recent commit (i.e. this month) - duplicate of issue #3548.

**Comment by TooWaBoo** [comment]
Need a bit help. Can someone give me three sceenshots like below from Linux, please? Because menus are windows too and I don't know what is rendered by a native window.
Use this theme, please:
[TooWaBlueTest-2.25-GTK3-20_.txt](https://github.com/Beep6581/RawTherapee/files/673376/TooWaBlueTest-2.25-GTK3-20_.txt)
![unbenannt](https://cloud.githubusercontent.com/assets/13800920/21487682/e8256478-cbcf-11e6-98ab-41da61b7bb61.PNG)
![unbenannt2](https://cloud.githubusercontent.com/assets/13800920/21487683/eb0ba6a2-cbcf-11e6-9b55-fc78fa113562.PNG)
![unbenannt3](https://cloud.githubusercontent.com/assets/13800920/21487684/ed2d43a0-cbcf-11e6-9fda-7f722e9c0c53.PNG)

**Comment by Beep6581** [comment]
As of now, please do not use the `gtk3-bugfix` branch anymore. It has been merged into `gtk3` and will soon be deleted.

```bash
git checkout gtk3
git pull
git branch -D gtk3-bugfix
```

Please use `master` for GTK2 builds and `gtk3` for GTK3 builds.

**Comment by Beep6581** [comment]
@Hombre57 as the scrollbar issue is a GTK+ bug, do we consider this to be solved and close it, or are you working on a workaround?

**Comment by dnw3039** [comment]
The scrollbar has been reduced in size since that I made my first comment
about it. As can be seen by the screen shot below it now obscures the  "
 :1  "  but leaves most of the reset arrow and the "  + " free to use. I
now have much less of a problem with it. So I would be happy to close the
issue.

(I am now on todays: Branch: gtk3 Version: 4.2.1492 Changeset:
4c20b1f4af1928d80467833b9f93cbb0ea26ca71 )


​
regards

dnw

**Comment by Beep6581** [comment]
@TooWaBoo https://filebin.net/is8btes6zjuq7odp

**Comment by TooWaBoo** [comment]
@Beep6581 
Thanks, looks identical, except the profile menu. There's no shadow.

**Comment by Hombre57** [comment]
@Beep6581 I'm not working on the Scrollbar bug nor the Curves' one and don't plane to do so. I didn't had time to post in Bugzilla yet, I'm very busy this week and will also be off for this w.e., my boss asked me to work at home for the whole w.e.

There is a Gtk3.24 out or around the corner. I don't think that it is fixed in that release though, bugzilla doesn't say so. You should probably create a tag to let everyone know that we're waiting on a dependency fix and leave the issue opened, otherwise someone will open a new one.

**Comment by Beep6581** [comment]
@Hombre57 then I will close this issue and refer people from the release notes or somewhere to the scrollbar issue #3545 and the curves issue #3525.

Thank you so much for all the time and effort you put into this!
